parlai/core/torch_generator_agent.py

```diff
@@ -1698,4 +1861,23 @@ def select_paths(self, logprobs, prior_scores, current_length):
         # Convert back to logspace.
         scores = sprobs[hyp_ids, choices].log()
         best_scores = prior_scores.expand_as(scores) + scores
-        return (hyp_ids, tok_ids, best_scores)
+
+        vocab_size = logprobs.size()[1]
+        ranks = logprobs.scatter(
+            1,
+            logprobs.argsort(descending=True),
+            torch.arange(vocab_size)
+            .type(torch.FloatTensor)
+            .to(logprobs.device)[None, :]
+            .repeat((self.beam_size, 1))
+            + 1,
+        )
+        tok_ranks = torch.gather(ranks, 1, tok_ids[None, :]).view(-1)
```

Comment:
same as in top-k, you are actually already sorting the probs above, so use that instead of re-sorting